pkg/scalers/rabbitmq_scaler.go

```diff
@@ -99,41 +100,42 @@ var rabbitmqLog = logf.Log.WithName("rabbitmq_scaler")
 
 // NewRabbitMQScaler creates a new rabbitMQ scaler
 func NewRabbitMQScaler(config *ScalerConfig) (Scaler, error) {
+	s := &rabbitMQScaler{}
+
+	metricType, err := GetMetricTargetType(config)
+	if err != nil {
+		return nil, fmt.Errorf("error getting scaler metric type: %s", err)
+	}
+	s.metricType = metricType
+
 	meta, err := parseRabbitMQMetadata(config)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing rabbitmq metadata: %s", err)
 	}
-	httpClient := kedautil.CreateHTTPClient(meta.timeout, false)
+	s.metadata = meta
+	s.httpClient = kedautil.CreateHTTPClient(meta.timeout, false)
 
-	if meta.protocol == httpProtocol {
-		return &rabbitMQScaler{
-			metadata:   meta,
-			httpClient: httpClient,
-		}, nil
-	}
+	if meta.protocol == amqpProtocol {
```

Comment:
```suggestion
	if meta.protocol != httpProtocol {
```

the protocol could even be `auto`

Reply:
I think it must be either `amqp` or `http` after [the parsing](https://github.com/kedacore/keda/blob/main/pkg/scalers/rabbitmq_scaler.go#L172-L176), but I don't mind changing it 😄

Reply:
Right, so disreagard then :)